.github/workflows/push-build.yaml

```
@@ -37,16 +40,24 @@ jobs:
           aws-region: us-east-1
 
       - name: unshallow
-        run: git fetch --prune --unshallow --tags
+        run: git fetch --prune --unshallow --tags --force
 
       # creating custom env var
       - name: set env
-        run: echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+        run: |
+          echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "GITHUB_REF"=${GITHUB_REF} >> $GITHUB_ENV
 
       - name: build and release
         env:
           GIT_API_TOKEN: ${{ secrets.GIT_API_TOKEN }}
           BRANCH_NAME: ${{ env.BRANCH_NAME }}
           AWS_USR: ${{ secrets.AWS_USR }}
           AWS_PSW: ${{ secrets.AWS_PSW }}
-        run: tests/scripts/build-release.sh
+          GITHUB_REF: $ {{ env.GITHUB_REF }}
+        run: |
+          if [[ ${GITHUB_REF} =~ master|v* ]]; then
+            tests/scripts/build-release.sh publish_and_promote
```

How about separate publish and promote calls like this?
```suggestion
            tests/scripts/build-release.sh publish
            tests/scripts/build-release.sh promote
```

---

we have `TAG_WITH_SUFFIX=true` so different calls will bring duplicate code

---

Could `publish` read an arg `$1` to decide if it should tag with suffix?

---

As discussed, let's keep it this way and this optimization could be a separate PR